models/src/main/protobuf/CasperMessage.proto

```diff
@@ -195,6 +195,11 @@ message ConsumeEvent {
 message CommEvent {
   ConsumeEvent consume = 1;
   repeated ProduceEvent produces = 2;
+  repeated Peek peeks = 3;
+}
+
+message Peek {
+  int32 channelIndex = 1;
```

Comment:
Why not just make it part of ConsumeEvent?
Actually we have to - peeks don't always come in comm-events, they can await their future pairs in the cold store as well.

Reply:
`they can await their future pairs in the cold store as well.` that's called a COMM :D

Reply:
1) There's no need to do that and it would make the Casper events model different than that of RSpace
2) This is why peeks are stored along with `WaitingContinuation`

Reply:
is there a reason to put this information in the event log? what will an unmatched consume that has a peeked channel mean to the merge strategy? last time this was discussed I think we arrived at the conclusion that it's enough that the TS knows about peek in consume

Reply:
I think I'm missing sth. How do we discern `4X` from` `PX` then?

Reply:
Are you sure we have to?

Reply:
Are you sure we don't?

Reply:
They behave identically in the mergeability matrix, so yes. What would be your reason for discerning them?

Reply:
That this decision should be taken in the code that implements the mergeability rules. Otherwise let's talk about what @dzajkowski just suggested and not expose peek to the merge rules at all.

Reply:
No, it doesn't need to be taken there. 
Not exposing peek to merge rules at all will make it impossible to discern `P!` from `4!` and `P!!` from `4!!`

Reply:
> No, it doesn't need to be taken there.

I agree that this PR could get in, but I'd like us three to meet and agree on an approach for having 'merge for peek' before we proceed any further.

> Not exposing peek to merge rules at all will make it impossible to discern P! from 4! and P!! from 4!!

It's fine though, as we'd restrict the MergeLog to things that change the TS.  P! and P!! don't.

BTW there's an error in the spreadsheet: `4! P!/!!` should merge, as we can always say that the P happened before the 4 in the 'merged trace'. This means that `P !/!!` merges always. @pyrocto please confirm :)

Reply:
> 4! P!/!! should merge, as we can always say that the P happened before the 4 in the 'merged trace'. This means that P !/!! merges always

This makes sense, good catch

Reply:
> 4! P!/!! should merge, as we can always say that the P happened before the 4 in the 'merged trace'.  This means that P !/!! merges always. @pyrocto please confirm :)

I agree.